standalone/prouter/src/main.rs

```diff
@@ -0,0 +1,512 @@
+    #[structopt(
+        default_value = "ws://localhost:9944",
+        long,
+        help = "Parachain rpc websocket endpoint"
+    )]
+    substrate_ws_endpoint: String,
+
+    #[structopt(long, help = "Don't wait the substrate nodes to sync blocks")]
+    no_wait: bool,
+
+    // Provide custom public endpoint. Will disable i2p router if this field is provided
+    #[structopt(long, help = "Specify the type of the endpoint", default_value = "i2p")]
+    endpoint_type: EndpointType,
+
+    #[structopt(
+        long,
+        help = "Provide custom public endpoint. E.g. http://xxx:3333"
+    )]
+    endpoint: Option<String>,
+
+    // PRouter Settings
+    #[structopt(long, help = "Auto restart self after an error occurred")]
+    auto_restart: bool,
+
+    #[structopt(
+        default_value = "10",
+        long,
+        help = "Max auto restart retries if it continuously failing. Only used with --auto-restart"
+    )]
+    max_restart_retries: u32,
+
+    // I2P Router Setting, Only effective with I2P endpoint
+    #[structopt(
+        long,
+        default_value = "600",
+        help = "Seconds of pRouter gracefully shutting down"
+    )]
+    shutdown_interval: u64,
+
+    #[structopt(long, default_value = "./prouter_data", help = "Path to store pRouter data")]
+    datadir: String,
+
+    #[structopt(long, help = "Override default i2pd config file provided")]
+    override_i2pd: Option<String>,
+
+    #[structopt(
+        long,
+        help = "Override default tunnels config file to provided. You need to prepare your own key file, path is relative to datadir"
+    )]
+    override_tun: Option<String>,
+
+    #[structopt(
+        default_value = "127.0.0.1",
+        long,
+        help = "Host for the public API, without `http://` prefix. Required to support http protocol"
+    )]
+    exposed_address: String,
+
+    #[structopt(
+        default_value = "8001",
+        long,
+        help = "Port for the public API. Required to support http protocol"
+    )]
+    exposed_port: u16,
+
+    #[structopt(
+    default_value = "127.0.0.1",
+    long,
+    help = "Host for the pRouter server, without `http://` prefix. Required to support http protocol"
+    )]
+    server_address: String,
+
+    #[structopt(
+    default_value = "8100",
+    long,
+    help = "Port for the pRouter server. Required to support http protocol"
+    )]
+    server_port: u16,
+}
+
+fn preprocess_path(path_str: &String) -> Result<PathBuf> {
+    // Check path exists
+    let path = Path::new(&path_str);
+    if !path.exists() {
+        fs::create_dir(&path)?;
+    }
+    // Convert to absolute path for i2pd config
+    let absolute_path = path
+        .canonicalize()
+        .expect("Any path should always be able to converted into a absolute path");
+
+    Ok(absolute_path)
+}
+
+// static API references for multi thread accessing in rocket server
+lazy_static! {
+    static ref PARA_API: Arc<Mutex<Option<ParachainApi>>> = Arc::new(Mutex::new(None));
```

Comment:
Since there is an API instance put into the Rocket managed state, why do we need a global one?